src/poetry/utils/extras.py

```diff
@@ -4,20 +4,20 @@
 
 
 if TYPE_CHECKING:
+    from collections.abc import Collection
     from collections.abc import Iterable
     from collections.abc import Iterator
-    from collections.abc import Sequence
     from typing import Mapping
 
     from packaging.utils import NormalizedName
     from poetry.core.packages.package import Package
 
 
 def get_extra_package_names(
-    packages: Sequence[Package],
+    packages: Iterable[Package],
     extras: Mapping[str, list[str]],
-    extra_names: Sequence[str],
-) -> Iterable[str]:
+    extra_names: Collection[str],
+) -> Iterable[NormalizedName]:
```

Comment:
Maybe, we should remove the type names from the docstring:
"A collection of packages, such as ..." -> "Packages, such as ..."
"A list of strings ..." -> "Strings ..."

(I wonder if the param descriptions add any value. Maybe not for `packages`, but maybe for `extras`.)

Reply:
I'm happy enough treating the types and the text as complementary documentation, it doesn't much bother me if the words are slightly imprecise - in the way that ordinary English often is - when we have the annotations making things explicit.

(But it wouldn't bother me if anyone did change this)